eng/pipelines/templates/stages/archetype-java-release-batch.yml

```diff
@@ -0,0 +1,497 @@
+                  # and then call to update_versions will then apply all of the updates.
+                  - ${{ each artifact in parameters.Artifacts }}:
+                    - script: |
+                        python --version
+                        python eng/versioning/set_versions.py --build-type ${{ parameters.SDKType }} --increment-version --ai ${{ artifact.name }} --gi ${{ artifact.groupId }}
+                      displayName: Increment package version for ${{ artifact.groupId }} ${{ artifact.name }}
+
+                  - script: |
+                      python --version
+                      python eng/versioning/update_versions.py --update-type library --build-type ${{ parameters.SDKType }} --avi --sr
+                    displayName: Update pom files for incremented versions
+
+                  - template: /eng/common/pipelines/templates/steps/create-pull-request.yml
+                    parameters:
+                      PRBranchName: increment-package-version-${{ parameters.ServiceDirectory }}-$(Build.BuildId)
+                      CommitMsg: "Increment package versions for ${{ parameters.ServiceDirectory }} releases"
+                      PRTitle: "Increment versions for ${{ parameters.ServiceDirectory }} releases"
+                      PRLabels: "auto-merge"
+                      CloseAfterOpenForTesting: '${{ parameters.TestPipeline }}'
+
+        # JRS - This needs to be updated with changes to batch up Docs.MS publishing otherwise
+        #       it'll take way too long and timeout for areas with large numbers of libraries.
+        - deployment: PublishDocsMs
+          displayName: Docs.MS Release
+          condition: and(succeeded(), ne(variables['Skip.PublishDocs'], 'true'))
+          environment: githubio
+          dependsOn: PublishPackage
+
+          pool:
+            name: azsdk-pool-mms-ubuntu-2004-general
+            vmImage: MMSUbuntu20.04
+
+          variables:
+            - template: /eng/pipelines/templates/variables/globals.yml
+            - name: DocValidationImageId
+              value: azuresdkimages.azurecr.io/javarefautocr:latest
+
+          strategy:
+            runOnce:
+              deploy:
+                steps:
+                  - download: current
+                    displayName: 'Download Artifact: ${{parameters.ArtifactName}}'
+                    artifact: ${{parameters.ArtifactName}}
+                  - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+                    parameters:
+                      SkipDefaultCheckout: true
+                      Paths:
+                        - sdk
+                        - common/perf-test-core
+                        - .github/CODEOWNERS
+                        - '!sdk/**/test-recordings'
+                        - '!sdk/**/session-records'
+                  - ${{ each artifact in parameters.Artifacts }}:
```

Comment:
I assume we plan to process these in an inner loop instead of one at a time.

Reply:
Not for this particular loop. This is exactly the same setup that @hallipr had added to the nightly docs publishing. I can let him speak to why this isn't batched but I'll say that the processing in this template is less than 3 seconds per library. I don't know if this is worth any turbulence at this time.

Reply:
For this, yes. That loop was moved down so the PackageInfoLocations is what's being set. I was waiting to mimic the changes @hallipr had made to release yml on Tuesday. I'd needed that set of changes in before I could make mine which were done in the first commit this morning.